xc/cancel_utils_test.ml

```diff
@@ -28,50 +26,27 @@ let xenstore_test xs _ =
     Thread.create
       (fun () ->
         Thread.delay 1. ;
-        Xenops_task.cancel tasks task.Xenops_task.id)
+        Xenops_task.with_cancel task (fun () -> ()) (fun () -> ()))
```

Comment:
`Fun.id`?

Reply:
The current function is more restrictive, type-wise. I like it better.